cmd/vulcan-snyk/main.go

```diff
@@ -0,0 +1,261 @@
+						vulcanVulnerability.CWEID = uint32(cweID)
+					}
+				}
+
+				for _, ref := range snykIssues[0].References {
+					vulcanVulnerability.References = append(vulcanVulnerability.References, ref.URL)
+				}
+
+				vulns = append(vulns, *vulcanVulnerability)
+			}
+		}
+
+		state.AddVulnerabilities(vulns...)
+
+		return nil
+	}
+	c := check.NewCheckFromHandler(checkName, run)
+
+	c.RunAndServe()
+}
+
+func createImpactDetails(vulnerabilities []SnykVulnerability) string {
+	res := ""
+	for _, vulnerability := range vulnerabilities {
+		str := "Introduced through: "
+		n := len(vulnerability.From)
+		for i := 0; i < n; i++ {
+			str = str + vulnerability.From[i]
+			if i < n-2 {
+				str = str + " > "
+			}
+		}
+		res = res + str + "\n"
+	}
+	return res
+}
+
+var regexpRemediationTagBegin = regexp.MustCompile(`(?i)<h2.*id="remediation".*</h2>`)
+var regexpRemediationTagEnd = regexp.MustCompile(`(?i)<h2.*>`)
+var bluemondayParser = bluemonday.StrictPolicy()
+
+func extractRecommendations(buf []byte) []string {
+	markdownParser := parser.NewWithExtensions(parser.CommonExtensions | parser.AutoHeadingIDs)
+
+	res := []string{}
+
+	bufStr := string(buf)
+	bufStr = strings.ReplaceAll(bufStr, "\\\\r\\\\n", "\n")
+	bufStr = strings.ReplaceAll(bufStr, "\\\\n", "\n")
+	bufStr = strings.ReplaceAll(bufStr, "--|", "---|")
+
+	html := markdown.ToHTML([]byte(bufStr), markdownParser, nil)
+
+	locationTagRemediations := regexpRemediationTagBegin.FindIndex(html)
+	if len(locationTagRemediations) > 1 {
+		remediationSection := html[locationTagRemediations[1]:]
+
+		locationTagRemediationsEnd := regexpRemediationTagEnd.FindIndex(remediationSection)
+		if len(locationTagRemediationsEnd) > 1 {
+			remediationSection = remediationSection[:locationTagRemediationsEnd[0]]
+		}
+
+		remediationStrLines := strings.Split(string(remediationSection), "\n")
+		for _, line := range remediationStrLines {
+			if len(line) > 0 {
+				aux := bluemondayParser.Sanitize(line)
+				aux = strings.Trim(aux, "\n")
+				if len(aux) > 0 {
+					res = append(res, aux)
+				}
+			}
+		}
+	}
+	return res
+}
+
+var regexpOverviewTagBegin = regexp.MustCompile(`(?i)<h2.*id="overview".*</h2>`)
+
+var regexpDetailsTagBegin = regexp.MustCompile(`(?i)<h2.*id="details".*</h2>`)
+
+var regexpNextH2TagBegin = regexp.MustCompile(`(?i)<h2`)
+
+func extractDetails(buf []byte) string {
+	markdownParser := parser.NewWithExtensions(parser.CommonExtensions | parser.AutoHeadingIDs)
+
+	bufStr := string(buf)
+	bufStr = strings.ReplaceAll(bufStr, "\\\\r\\\\n", "\n")
+	bufStr = strings.ReplaceAll(bufStr, "\\\\n", "\n")
+	bufStr = strings.ReplaceAll(bufStr, "--|", "---|")
+
+	html := markdown.ToHTML([]byte(bufStr), markdownParser, nil)
+
+	locationTagDetails := regexpDetailsTagBegin.FindIndex(html)
+	if len(locationTagDetails) > 1 {
+		html = html[locationTagDetails[1]:]
+		locationTagNextH2 := regexpNextH2TagBegin.FindIndex(html)
+		if len(locationTagNextH2) > 1 {
+			html = html[:locationTagNextH2[0]]
```

Comment:
If you prefer using regex here, I would just capture the content between tags instead of working by index, like this:
```
var regexpOverviewTag = regexp.MustCompile(`(?i)<h2.*?id="overview">(.*?)</h2>`)
return regexpOverviewTag.FindStringSubmatch(html)[1]
```
I think that is easier to understand.